oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/FlatFileStoreUtils.java

```diff
@@ -30,46 +32,67 @@
+        switch (compressionType) {
+            case COMPRESSION_TYPE_LZ4:
+                return ExternalSort.CompressionType.LZ4;
+            case COMPRESSION_TYPE_GZIP:
+                return ExternalSort.CompressionType.GZIP;
+            case COMPRESSION_TYPE_NONE:
+                return ExternalSort.CompressionType.NONE;
+            default:
+                return ExternalSort.CompressionType.NONE;
+        }
+    }
+
     public static BufferedReader createReader(File file, boolean compressionEnabled) {
+        return createReader(file, compressionEnabled ? COMPRESSION_TYPE_GZIP : COMPRESSION_TYPE_NONE);
+    }
+
+    public static BufferedReader createReader(File file, String compressionType) {
         try {
-            BufferedReader br;
             InputStream in = new FileInputStream(file);
-            if (compressionEnabled) {
-                br = new BufferedReader(new InputStreamReader(new GZIPInputStream(in, 2048), UTF_8));
-            } else {
-                br = new BufferedReader(new InputStreamReader(in, UTF_8));
-            }
-            return br;
+            return new BufferedReader(new InputStreamReader(getExternalSortCompressionType(compressionType).getInputStream(in)));
         } catch (IOException e) {
             throw new RuntimeException("Error opening file " + file, e);
         }
     }
 
     public static BufferedWriter createWriter(File file, boolean compressionEnabled) throws IOException {
+        return createWriter(file, compressionEnabled ? COMPRESSION_TYPE_GZIP : COMPRESSION_TYPE_NONE);
+    }
+
+    public static BufferedWriter createWriter(File file, String compressionType) throws IOException {
         OutputStream out = new FileOutputStream(file);
-        if (compressionEnabled) {
-            out = new GZIPOutputStream(out, 2048) {
-                {
-                    def.setLevel(Deflater.BEST_SPEED);
-                }
-            };
-        }
-        return new BufferedWriter(new OutputStreamWriter(out, UTF_8));
+        return new BufferedWriter(new OutputStreamWriter(getExternalSortCompressionType(compressionType).getOutputStream(out)));
     }
 
     public static long sizeOf(List<File> sortedFiles) {
         return sortedFiles.stream().mapToLong(File::length).sum();
     }
 
-    public static String getSortedStoreFileName(boolean compressionEnabled){
-        return compressionEnabled ? "store-sorted.json.gz" : "store-sorted.json";
+    public static String getSortedStoreFileName(boolean compressionEnabled) {
+        return getSortedStoreFileName(compressionEnabled ? COMPRESSION_TYPE_GZIP : COMPRESSION_TYPE_NONE);
+    }
+
+    public static String getSortedStoreFileName(String compressionType) {
+        String name = "store-sorted.json";
+        if (compressionType.equals(COMPRESSION_TYPE_NONE)) {
+            return name;
+        }
+        return name + "." + compressionType;
```

Comment:
here, we could use CompressionType.toString().toLowerCase(Locale.ENGLISH) -- Locale.ENGLISH is important because in Turkey, sorry Türkiye, "GZIP" would be converted to "gzıp". Notice here, the "i" is missing the dot (it's a dotless i). Yes it's weird, it's the "Turkish Locale Problem".

We could add a method "getSuffix" to the CompressionType enum, and hardcode the types there, that might be better. Then we can also use ".gzip" and ".lz4" and "" (empty string for none).